### PR TITLE
Update outdated AgentCheck test

### PR DESCRIPTION
**Pull request by nmuesch**
### What does this PR do?

Updates the old tests that were created when AgentCheck was in the datadog-agent repo. 

### Motivation

Now AgentCheck is in datadog_checks_base and the import is redirected there. This PR https://github.com/DataDog/integrations-core/pull/2293 changed the behavior so we aren't throwing an exception on a single bad metric. Currently the CI is failing without this change as the old test expects an exception. 

The file wasn't removed because this is run as a check in another test that validates information about the CheckID. 

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/collector/py/tests/testaggregator.py [hunk 1]
```diff
@@ -22,12 +22,7 @@ def check(self, instance):
         self.gauge("testmetric", 0, tags=None)
         self.gauge("testmetricnonevalue", None) # metrics with None values should be ignored by AgentCheck
         self.gauge("testmetricstringvalue", "2") # string values should be cast to floats
-        try:
-            self.gauge("testmetricstringvalue", "notcastabletofloat") # values not castable to floats should raise an exception
-        except ValueError:
-            pass
-        else:
-            raise Exception("Expected gauge to raise ValueError")
+        self.gauge("testmetricstringvalue", "notcastabletofloat") # values not castable to floats should raise an exception
```
Comment (arbll) [inline pkg/collector/py/tests/testaggregator.py]: `# values not castable to floats should raise an exception` is no longer true right ?

Comment (nmuesch) [reply pkg/collector/py/tests/testaggregator.py]: Good catch, updated

**Review by arbll — APPROVED** [review]
🔧

**Review by masci — APPROVED** [review]
🐍

**Pull request merged**